crates/matrix-sdk-crypto/src/machine.rs

```diff
@@ -901,12 +920,17 @@ impl OlmMachine {
         self.account.update_key_counts(one_time_key_count, unused_fallback_keys).await;
     }
 
-    async fn handle_to_device_event(&self, event: &ToDeviceEvents) {
+    async fn handle_to_device_event(
+        &self,
+        changes: &mut Changes,
+        event: &ToDeviceEvents,
+    ) -> OlmResult<()> {
```

Comment:
Since `add_withheld_info` doesn't need to be fallible, this doesn't need to either.